assets/src/edit-story/components/form/textArea.js

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+const StyledTextArea = styled.textarea`
+  width: 100%;
+  padding: 0;
+  border: none;
+  appearance: none;
+  box-shadow: none !important;
+  outline: none;
+  background-color: transparent;
+  color: ${({ theme }) => theme.colors.fg.white};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  resize: ${({ resizeable }) => (resizeable ? 'auto' : 'none')};
+
+  &:disabled {
+    background-color: transparent;
+    color: ${({ theme }) => theme.colors.fg.white};
+  }
+`;
+
+const Container = styled.div`
+  width: 100%;
+  color: ${({ theme }) => rgba(theme.colors.fg.white, 0.3)};
+  background-color: ${({ theme }) => rgba(theme.colors.bg.black, 0.3)};
+
+  ${({ disabled, readOnly }) => (disabled || readOnly) && `opacity: 0.3`};
+
+  padding: 6px 12px;
+  padding-right: 6px;
+  border-radius: 4px;
+  border: 1px solid transparent;
+  &:focus-within {
+    border-color: ${({ theme }) => theme.colors.whiteout} !important;
+  }
+`;
+
+const Counter = styled.div`
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.tab.size};
+  text-align: right;
+  line-height: 1;
+  padding-right: 6px;
+`;
+
+function TextArea({
+  className,
+  placeholder,
+  value,
+  maxLength,
+  readOnly,
+  disabled,
+  resizeable,
+  showTextLimit,
+  rows,
+  onTextChange,
+  onBlur,
+  ...rest
+}) {
+  const hasMaxLength = useMemo(() => typeof maxLength === 'number', [
+    maxLength,
+  ]);
+
+  const showCounter = useMemo(() => showTextLimit && hasMaxLength, [
+    showTextLimit,
+    hasMaxLength,
+  ]);
+
+  const handleChange = useCallback(
+    (e) => {
+      const str = e.target.value || '';
+      const text = hasMaxLength ? str.slice(0, maxLength) : str;
+
+      onTextChange(text, e);
+    },
+    [onTextChange, hasMaxLength, maxLength]
+  );
+
+  const handleBlur = useCallback(
+    (e) => {
+      if (e.target.form) {
+        e.target.form.dispatchEvent(
+          new window.Event('submit', { cancelable: true })
+        );
+      }
+
+      if (onBlur) {
+        onBlur(e);
+      }
+    },
+    [onBlur]
+  );
+
+  return (
+    <Container className={className}>
+      <StyledTextArea
+        placeholder={placeholder}
+        maxLength={maxLength}
+        disabled={disabled}
+        readOnly={readOnly}
+        resizeable={resizeable}
+        rows={rows}
```

Comment:
All these properties could just be handled by `{...rest}` so we don't need to define them nor set defaults for them at all. One could argue that `value` could be too, but I find it important enough (one could argue critically) to include explicitly.

The default for `rows` could be set by explicitly setting it in the definition above as:

```js
const StyledTextArea = styled.textarea.attrs({ rows: 2 })`
```

And then `{ ...rest }` could properly overwrite where applicable.

Reply:
I agree with keeping the props explicit rather than being spread, makes it easier to read and understand in my opinion.

Reply:
We then need to keep adding props whenever we want to add new attributes to the `<textarea />` (e.g. `minLength`, `autocomplete`, `required`, `spellcheck`, `wrap`, etc).

For React components mostly just wrapping native HTML elements (and `<input />`s in particular), it's fairly common to only explicitly define the required, manipulated, or non-standard properties and leave all the "standard" HTML attributes as open-ended with custom spreading.

It's also the approach we've used [elsewhere](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/form/numeric.js#L155) in this project - but I'm definitely open to changing that, if that's the consensus.

Reply:
I feel a combination of both approaches for wrapping HTML elements is best.  I explicitly defined the props I feel would be the most used when using the `textArea`.  

- `rows` is useful for controlling the height, but I didn't define `cols` because most likely that will be handled with `width: 100%`.  
- `placeholder`, `maxLength`, `value`: are the 3 that will probably be used on every implementation.
- `readOnly`, `disabled`, `resizable`: I included these because I can imagine UI situations where each one could be enabled so I wanted to call them out for ease.
- `onBlur`: I wanted to call it out since it gets slightly overridden.

The remaining props included with {...rest} I believe are ones that will less likely get used, but could still be useful.

**TDLR;** I feel if we define the props that cover 90% of the use cases and allow {...rest} to handle the remaining edge cases, we'd get best of both worlds and that's what I tried to do with `textArea`.